sqlmesh/core/state_sync/base.py

```diff
@@ -158,17 +159,30 @@ def get_versions(self, validate: bool = True) -> Versions:
         """
         versions = self._get_versions()
 
-        def raise_error(lib: str, local: str | int, remote: str | int, ahead: bool = False) -> None:
+        def raise_error(
+            lib: str,
+            local: str | int,
+            remote: str | int,
+            remote_package_version: t.Optional[str] = None,
+            ahead: bool = False,
+        ) -> None:
             if ahead:
                 raise SQLMeshError(
-                    f"{lib} (local) is using version '{local}' which is ahead of '{remote}' (remote). Please run a migration ('sqlmesh migrate' command)."
+                    f"{lib} (local) is using version '{local}' which is ahead of '{remote}' (remote). "
+                    "Please run a migration ('sqlmesh migrate' command)."
                 )
             raise SQLMeshError(
-                f"{lib} (local) is using version '{local}' which is behind '{remote}' (remote). Please upgrade {lib} ('pip install --upgrade \"{lib.lower()}=={remote}\"' command)."
+                f"{lib} (local) is using version '{local}' which is behind '{remote}' (remote). "
+                f"Please upgrade {lib} ('pip install --upgrade \"{lib.lower()}=={remote_package_version or remote}\"' command)."
```

Comment:
Why even fallback on `remote`? Isn't it meaningless?

Reply:
It's used when SQLGlot local version is behind

Reply:
We could be explicit though and use the `remote_package_version` instead of falling back to `remote`.

Reply:
ah, I see

Reply:
it's really confusing that we use `remote_package_version` for sqlmesh and `remote` for sqlglot

Reply:
tried to clean this up here: https://github.com/TobikoData/sqlmesh/pull/1638/commits/f6a789795901423736e03e72ebfa9bd607ac8b34